### PR TITLE
Jdp230502 21

### PR DESCRIPTION
**Pull request by sew-k**
- finished Product entity with relations to group
- tests for Product save and delete with Group added
- modifications in ProductMapper

Last commit is related to need to perform full tests for ProductController in Postman - necessary modifications of Cart entity included.

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/CartController.java]: import com.kodilla.ecommercee.mapper.CartMapper and import com.kodilla.ecommercee.service.UserService not needed anymore

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/CartController.java]: You dumped the **get** for the whole list - I understand that it is simply unnecessary for the logic of the program and at this stage we find that it will not be useful to us (for example for tests?).

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/CartController.java:28]: Maybe we should throw an exception here?

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/CartController.java:33]: I think we should throw an exception.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/CartController.java:38]: The same - I think we should throw an exception here.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/CartController.java]: It's not a problem but it seems more natural to me to swap the order:
"/createOrder/{userId}"

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: java.util.ArrayList not needed anymore...

File: src/main/java/com/kodilla/ecommercee/domain/Order.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/domain/Order.java:1]: I didn't catch it earlier, and since it's impossible to tag already existing, approved code, I write in a general class comment: 
```
    public Order(OrderStatus orderStatus, Cart cart, **User user**) {
        this.orderStatus = orderStatus;
        this.cart = cart;
    }
```
User is not used for anything here, so we should remove it.

File: src/main/java/com/kodilla/ecommercee/domain/Product.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/domain/Product.java]: import java.util.ArrayList and import javax.validation.constraints.NotNull not needed anymore...

File: src/main/java/com/kodilla/ecommercee/dto/CartDto.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/dto/CartDto.java]: I'm not sure - how does this relate to the JSON specification for cart?

File: src/main/java/com/kodilla/ecommercee/dto/ItemDto.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/dto/ItemDto.java:1]: I think we've decided that ItemDto was going to be deleted - it's not tied to anything, it's not used for anything.

File: src/main/java/com/kodilla/ecommercee/exception/CartNotFoundException.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/exception/CartNotFoundException.java:1]: We need to hook this exception to calls in controller - it hasn't been used yet.

File: src/test/java/com/kodilla/ecommercee/service/ProductServiceTestSuite.java
Comment (WorstWorm) [inline src/test/java/com/kodilla/ecommercee/service/ProductServiceTestSuite.java:1]: Please remove
import com.kodilla.ecommercee.repository.GroupRepository and import org.junit.jupiter.api.BeforeAll;

File: src/main/java/com/kodilla/ecommercee/service/ProductService.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/service/ProductService.java]: so import org.springframework.beans.factory.annotation.Autowired is not needed anymore...

File: src/main/java/com/kodilla/ecommercee/service/OrderService.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/service/OrderService.java]: Not needed import...

File: src/main/java/com/kodilla/ecommercee/service/OrderService.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/service/OrderService.java:1]: I don't know how it got out of order...
Deletion request:
import com.kodilla.ecommercee.domain.OrderStatus;

File: src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java:7]: Please cleanup imports:
- double import com.kodilla.ecommercee.domain.Group (both not needed)
- import com.kodilla.ecommercee.repository.GroupRepository;
- import com.kodilla.ecommercee.repository.ProductRepository;

File: src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java:5]: With this changes 
import com.kodilla.ecommercee.domain.Order;
import com.kodilla.ecommercee.domain.User;
not needed anymore...

File: src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java:1]: The question of this compliance with the specification because it will affect the mapper to...

**Review by WorstWorm — COMMENTED** [review]
A lot of small fixes (mainly imports - let's not add redundant ones anymore) and bigger - specification problem.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java]: I have to had to remove it accidentaly. But if you insist, I can restore it of course.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java:28]: Yes, but subject of this task was Product entity and ProductController. I think this should be done in other pull - requests.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java:33]: Same as above

File: src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java:7]: Fixed in next commit.

**Review by JakubFir — APPROVED** [review]

**Review by WorstWorm — APPROVED** [review]
I don't want to slow down the work at this stage, so I tick CR but I am asking you to review the imports in individual classes before the merge - I still see that there are unused imports in the product class waiting for the merge, and maybe there are more of them. Intellij itself points them out in class warnings.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java:28]: I don't know if it's worth creating a separate pull request for this. Tasks don't necessarily have to be 1-1 on PR. It's a team work tool and not something that is in any way important "outside". But It's your call...

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (WorstWorm) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java]: Your decision, but it will be useful for testing anyway.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java]: Removed

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java]: I see on master now Jakub changed to:
    @GetMapping("{userId}")
    public List<ProductDto> getCart(@PathVariable("userId") Long userId) {
        return cartMapper.mapToCartDtoProducts(cartDbService.getUserCart(userId));
    }
- I think its the right way for now.

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java:28]: Added exception throwing (UserNotFoundException)

File: src/main/java/com/kodilla/ecommercee/controller/CartController.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/controller/CartController.java:38]: Added

File: src/main/java/com/kodilla/ecommercee/service/OrderService.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/service/OrderService.java]: Removed

File: src/main/java/com/kodilla/ecommercee/service/OrderService.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/service/OrderService.java:1]: Removed

File: src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java:5]: Removed

**Pull request merged**
